cpp-examples/demos/crypto.cc

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+#include <charconv>
+#include <exception>
+#include <iostream>
+#include <sstream>
+#include "deephaven/client/client.h"
+#include "deephaven/client/chunk/chunk.h"
+#include "deephaven/client/column/column_source.h"
+#include "deephaven/client/utility/utility.h"
+
+using deephaven::client::Client;
+using deephaven::client::TableHandle;
+using deephaven::client::TableHandleManager;
+using deephaven::client::chunk::Int64Chunk;
+using deephaven::client::column::Int64ColumnSource;
+using deephaven::client::container::RowSequence;
+using deephaven::client::table::Table;
+using deephaven::client::utility::verboseCast;
+
+namespace {
+void mainMenu(const TableHandleManager &manager);
+
+// demo options
+void printFull(const TableHandleManager &manager);
+void printDiffs(const TableHandleManager &manager);
+
+// utilities
+int64_t readNumber(std::string_view prompt);
+std::string readString(std::string_view prompt);
+
+// a bunch of work to create an ostream adaptor
+template<typename ARROW_OPTIONAL>
```

Comment:
I think you can delete this class `OptionalAdaptor` and the method below `adaptOptional`